pil-analyzer/src/type_inference.rs

```diff
@@ -671,7 +668,7 @@ impl<'a> TypeChecker<'a> {
     /// Returns the type expected at statement level, given the current function context.
     fn statement_type(&self) -> ExpectedType {
         if self.lambda_kind == FunctionKind::Constr {
-            self.constr_function_statement_type.clone()
+            constr_function_statement_type()
```

Comment:
If we allow this expected_type to be a builtin instead of coming from the typechecker itself, `self.constr_function_statement_type` is never read and can be deleted along with its lifetime `'a`


Of course, the price is that expected_type is no longer parameterised in the type_checker. What do you think?

Reply:
I mean yeah, that was the idea. It doesn't need to be parameterized because it will always be the same type. The main idea of moving it out was that the type checker should remain "pure" and all the builtins and the specific stuff should be somewhere else.

Reply:
Perfect. Then it's ready for review :)